libraries/stdlib/test/ranges/RangeTest.kt

```diff
@@ -5,11 +5,198 @@
 
 package test.ranges
 
+import kotlin.math.min
+import kotlin.math.sign
 import kotlin.test.*
 
+fun <T : Comparable<T>, S> testProgressionCollection(
+    limitRange: Collection<T>,
+    stepRange: Iterable<S>,
+    makeRange: (T, T, S) -> Collection<T>,
+    nextItem: (T, S) -> T?,
+    sign: (S) -> Int
+) {
+    for (start in limitRange)
+        for (finish in limitRange)
+            for (step in stepRange) {
```

Comment:
This amount of iterations times out when running in JS. 
For example, in `longProgressionCollection` test, it's invoked with arguments that makes it run `21x21x20x(~20+20*~10+20+100*4*~2)=~9mln` elementary operations with rather high memory pressure.
Any ideas how to make it more moderate?